### PR TITLE
Aligned module versions  to 1.4-SNAPSHOT

### PR DESCRIPTION
**Pull request by offtherailz**
## Description
Aligned versions of mapstore modules.
**Please check if the PR fulfills these requirements**
- [ ] The commit message follows our guidelines: https://github.com/geosolutions-it/MapStore2/blob/master/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x", remove the others)
 - [ ] Bugfix
 - [ ] Feature
 - [ ] Code style update (formatting, local variables)
 - [ ] Refactoring (no functional changes, no api changes)
 - [ ] Build related changes
 - [ ] CI related changes
 - [ ] Other... Please describe:

## Issue

**What is the current behavior?**

#<issue>

**What is the new behavior?**

## Breaking change
**Does this PR introduce a breaking change?** (check one with "x", remove the other)
 - [ ] Yes, and I documented them in migration notes
 - [x] No

### CODE REVIEW

**Comment by offtherailz** [comment]
I've noticed that printing modules are not updated. 

In release procedure we have : 
```
mvn versions:set -DnewVersion=1.4-SNAPSHOT -DgenerateBackupPoms=false
```
I noticed that this doesn't update the modules included only in some profiles, `printing` and so on. Anyway adding the profiles the things doesn't change. 

I had to run this to update all modules. 

```
mvn release:update-versions -DdevelopmentVersion=1.4-SNAPSHOT -Pprinting,printingbundle,release
```
But doing this I noticed that somewhere we use project properties. 

@taba90 what it could be the better approach to keep module versions in sync? What are the strategies used for other java projects we handle.

**Review by tdipisa — APPROVED** [review]

**Comment by tdipisa** [comment]
@taba90 can you double check this PR with a review^

File: release/bin-war/pom.xml [hunk 1]
```diff
@@ -1,10 +1,9 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0"
-    xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
     <modelVersion>4.0.0</modelVersion>
     <groupId>it.geosolutions.mapstore</groupId>
     <artifactId>mapstore-bin-war</artifactId>
     <packaging>war</packaging>
-    <version>${mapstore2.version}</version>
```
Comment (taba90) [inline release/bin-war/pom.xml:7]: then I guess that this property should go away https://github.com/geosolutions-it/MapStore2/blob/master/java/web/pom.xml#L13 so we avoid keeping it and creating confusion

**Review by taba90 — CHANGES_REQUESTED** [review]
Looks good to me. Just a minor note.
Regarding the question from @offtherailz I'm don't know actually a way to have the version set to the printing module in a simpler way.
Other java projects doesn't have all this modules with a root pom as it is the case of MapStore2, but instead they've a single root pom module and all the others as children of it.
Will add this as something to keep in mind for the incoming refactoring of mapstore dependecies and modules management.

**Review by taba90 — APPROVED** [review]

File: release/pom.xml [hunk 2]
```diff
@@ -1,10 +1,9 @@
-<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
-         xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/maven-v4_0_0.xsd">
   <modelVersion>4.0.0</modelVersion>
   <groupId>it.geosolutions.mapstore</groupId>
   <artifactId>mapstore-bin-release</artifactId>
   <packaging>pom</packaging>
-  <version>${mapstore2.version}</version>
+  <version>1.4-SNAPSHOT</version>
```
Comment (offtherailz) [inline release/pom.xml:6]: I think the release need to keep this version. The binary should generate the proper zip package with the version passed.

```suggestion
  <version>${mapstore2.version}</version>
```

File: release/bin-war/pom.xml [hunk 3]
```diff
@@ -1,10 +1,9 @@
...
+    <version>1.4-SNAPSHOT</version>
```
Comment (offtherailz) [inline release/bin-war/pom.xml:6]:
```suggestion
    <version>${mapstore2.version}</version>
```

**Comment by offtherailz** [comment]
@taba90 I re-checked. The two with the `mapstore.version` need to be set as they are, in order to obtain the proper version in the release package :
e.g. 

```
./build.sh XXXX release # XXXX can be 2022.01.01, for instance. 
[...]
[INFO] ------------------------------------------------------------------------
[INFO] Reactor Summary:
[INFO] 
[INFO] MapStore 2 - Backend Services 1.4-SNAPSHOT ......... SUCCESS [  3.492 s]
[INFO] MapStore 2 - WAR 1.4-SNAPSHOT ...................... SUCCESS [  4.958 s]
[INFO] MapStore 2 1.4-SNAPSHOT ............................ SUCCESS [  0.005 s]
[INFO] MapStore Product Web Application 1.4-SNAPSHOT ...... SUCCESS [  7.529 s]
[INFO] MapStore 2 Release Module WAR XXXX ................. SUCCESS [  6.844 s]
[INFO] MapStore 2 Release Module XXXX ..................... SUCCESS [ 33.853 s]
[INFO] MapStore Root 1.4-SNAPSHOT ......................... SUCCESS [  0.015 s]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  56.914 s
[INFO] Finished at: 2022-05-11T10:13:25+02:00

# creates --> release/target/mapstore2-XXXX-bin.zip 
```

**Pull request merged**